### PR TITLE
More Renaming in query

### PR DESCRIPTION
**Pull request by rakshith91**
Fixes https://github.com/Azure/azure-sdk-for-python/issues/20187

### CODE REVIEW

File: sdk/monitor/azure-monitor-query/azure/monitor/query/_models.py [hunk 1]
```diff
@@ -482,7 +485,9 @@ def _from_generated(cls, generated):
             unit=generated.unit,
             timeseries=[
                 TimeSeriesElement._from_generated(t) for t in generated.timeseries # pylint: disable=protected-access
-                ]
+                ],
+            display_description=generated.display_description,
+            error= error,
```
Comment (rakshith91) [inline sdk/monitor/azure-monitor-query/azure/monitor/query/_models.py:490]:
```suggestion
```

File: sdk/monitor/azure-monitor-query/CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +18,8 @@
 - `metric_namespace_name` is renamed to `fully_qualified_namespace`
 - `is_dimension_required` is renamed to `dimension_required`
 - `time_grain` is renamed to `granularity`
+- `interval` is renamed to `granularity`
```
Comment (yunhaoling) [inline sdk/monitor/azure-monitor-query/CHANGELOG.md:21]: I might lose the necessary context -- so `interval` and `time_grain` are the same thing now?

File: sdk/monitor/azure-monitor-query/CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,8 @@
 
 ### Features Added
 
+- Added addition `display_description` attribute to the `Metric` type.
```
Comment (yunhaoling) [inline sdk/monitor/azure-monitor-query/CHANGELOG.md:7]: super nit? 
```suggestion
- Added additional `display_description` attribute to the `Metric` type.
```

File: sdk/monitor/azure-monitor-query/CHANGELOG.md (hunk 2 above)
Comment (rakshith91) [reply sdk/monitor/azure-monitor-query/CHANGELOG.md:21]: yes - they are the same thing

Comment (rakshith91) [reply sdk/monitor/azure-monitor-query/CHANGELOG.md:21]: we just ended up renaming them differently in different places of swagger somehow

Comment (rakshith91) [inline sdk/monitor/azure-monitor-query/CHANGELOG.md:21]:
```suggestion
- `interval`  and `time_grain` are renamed to `granularity`
```

File: sdk/monitor/azure-monitor-query/CHANGELOG.md [hunk 4]
```diff
@@ -16,6 +18,8 @@
 - `metric_namespace_name` is renamed to `fully_qualified_namespace`
 - `is_dimension_required` is renamed to `dimension_required`
 - `time_grain` is renamed to `granularity`
```
Comment (rakshith91) [inline sdk/monitor/azure-monitor-query/CHANGELOG.md:20]:
```suggestion
```

**Review by yunhaoling — APPROVED** [review]

**Pull request merged**